pkg/system/ip.go

```diff
@@ -174,7 +180,14 @@ func (i *ipManager) Compare(configured, desired []string) ([]string, []string) {
 			}
 		}
 		if !found {
-			removals = append(removals, caddr)
+			if !v6 && isV4Addr(caddr) {
+				removals = append(removals, caddr)
+				continue
```

Comment:
without the bool condition the two `compare` functions thrash against each other.

The `desired` set is now passed into  this function in 2 places: with the v4 configset, and the v6 configset. So, anything not in `desired` is considered a removal - with the code as previously written, the v4 `compare` would delete all the 

So this follows the pattern we have used in this PR where we expose to methods on the interface that pass in a bool to the ultimate compare function:

```
func (i *ipManager) Compare4(configured, desired []string) ([]string, []string) {
	return i.Compare(configured, desired, false)
}

func (i *ipManager) Compare6(configured, desired []string) ([]string, []string) {
	return i.Compare(configured, desired, true)
}
```

Thoughts?

Reply:
The continue was to drop out of the loop instead of of doing the next case. I turned it into an else-if now. Will integration test new statement:

```
if !v6 && isV4Addr(caddr) {
  removals = append(removals, caddr)
} else if v6 && !isV4Addr(caddr) {
  removals = append(removals, caddr)
}
```